### DISCUSSION TITLE
How to send data from server process to client

### DISCUSSION THREAD

**Discussion by adriaandotcom** [nuxt/nuxt#17546] [Questions] [answered]
We collect some data from the server process that we can't run in the client.

For example, the country code from an IP address https://github.com/nuxt/framework/discussions/5115

How can I send this data to the client?

**Comment by TheAlexLichter** [comment, marked as answer]
You can use `useState` inside of a Nuxt plugin to transfer state over to the client. An example can be found in the [supabase module](https://github.com/nuxt-community/supabase-module/blob/main/src/runtime/plugins/supabase.server.ts) using a custom composable built on top of `useState`.
